### PR TITLE
topology: add support for plugins

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

File: agent/probes.go [hunk 1]
```diff
@@ -104,5 +106,33 @@ func NewTopologyProbeBundleFromConfig(g *graph.Graph, n *graph.Node) (*probe.Pro
...
+
+	for _, so := range list {
+		plugin, err := plugin.Open(path.Join(pluginsDir, so+".so"))
+		if err != nil {
+			logging.GetLogger().Errorf("Failed to load plugin %s: %s", so, err)
+			continue
+		}
+		symbol, err := plugin.Lookup("NewProbe")
+		if err != nil {
+			logging.GetLogger().Errorf("Not compliant plugin: %s", err)
```
Comment (hunchback) [inline agent/probes.go]: s/Not/Non/

**Comment by hunchback** [comment]
@lebauce - for now this is good but in future we may want to extend this such that a plugin would contain both a backend compiled shared-object and a UI bundle including JS, CSS and IMG files needed.

**Comment by lebauce** [comment]
run tests

**Comment by lebauce** [comment]
run skydive-compile-tests
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
Please review / merge

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Pull request merged**